src/main/java/seedu/address/model/person/Email.java

```diff
@@ -10,26 +10,20 @@
 public class Email {
 
     private static final String SPECIAL_CHARACTERS = "+_.-";
-    public static final String MESSAGE_CONSTRAINTS = "Emails should be of the format local-part@domain "
-            + "and adhere to the following constraints:\n"
-            + "1. The local-part should only contain alphanumeric characters and these special characters, excluding "
-            + "the parentheses, (" + SPECIAL_CHARACTERS + "). The local-part may not start or end with any special "
-            + "characters.\n"
-            + "2. This is followed by a '@' and then a domain name. The domain name is made up of domain labels "
-            + "separated by periods.\n"
+    public static final String MESSAGE_CONSTRAINTS = "Emails should be of the format local-part@domain.tld and adhere"
+            + "to the following constraints:\n"
+            + "1. The local-part should only contain alphanumeric characters and these special characters ("
+            + SPECIAL_CHARACTERS + "). The local-part may not start or end with any special characters.\n"
+            + "2. This is followed by a '@' , a domain name, a '.', and the top-level domain (tld). "
+            + "The domain name is made up of domain labels separated by periods.\n"
             + "The domain name must:\n"
-            + "    - end with a domain label at least 2 characters long\n"
+            + "    - end with a top-level domain (tld) which is at least 2 characters long\n"
             + "    - have each domain label start and end with alphanumeric characters\n"
-            + "    - have each domain label consist of alphanumeric characters, separated only by hyphens, if any.";
-    // alphanumeric and special characters
-    private static final String ALPHANUMERIC_NO_UNDERSCORE = "[^\\W_]+"; // alphanumeric characters except underscore
-    private static final String LOCAL_PART_REGEX = "^" + ALPHANUMERIC_NO_UNDERSCORE + "([" + SPECIAL_CHARACTERS + "]"
-            + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_PART_REGEX = ALPHANUMERIC_NO_UNDERSCORE
-            + "(-" + ALPHANUMERIC_NO_UNDERSCORE + ")*";
-    private static final String DOMAIN_LAST_PART_REGEX = "(" + DOMAIN_PART_REGEX + "){2,}$"; // At least two chars
-    private static final String DOMAIN_REGEX = "(" + DOMAIN_PART_REGEX + "\\.)*" + DOMAIN_LAST_PART_REGEX;
-    public static final String VALIDATION_REGEX = LOCAL_PART_REGEX + "@" + DOMAIN_REGEX;
+            + "    - have each domain label and tld consist of alphanumeric characters,"
+            + " separated only by hyphens, if any.";
+    // Credit: OWASP Validation Regex Repository
```

Comment:
Good job crediting the source! I believe this does count as code reuse though, so remember to declare code reuse on Canvas.

Reply:
Thanks for pointing that out, will do that.